carsus/io/cmfgen/base.py

```diff
@@ -809,3 +838,116 @@ def _get_levels_lines(self, data):
+            collisions = data_dict["collisions"].copy()
+            levels_combine = self.levels.copy().reset_index()
+
+            label_ind_mapping = {
+                label: index for label, index in zip(levels_combine.label, levels_combine.level_index)
+            }
+            
+            label_g_mapping = {
+                label: g for label, g in zip(levels.label, levels.g)
+            }
+
+            gi, lower_level_index, upper_level_index = [], [], []
+
+            for ll, ul in zip(collisions.label_lower, collisions.label_upper):
+                if ll in label_ind_mapping:
+                    lower_level_index.append(label_ind_mapping[ll])
+                else:
+                    lower_level_index.append(np.nan)
+                
+                if ll in label_g_mapping:
+                    gi.append(label_g_mapping[ll])
+                else:
+                    gi.append(np.nan)
+
+                if ul in label_ind_mapping:
+                    upper_level_index.append(label_ind_mapping[ul])
+                else:
+                    upper_level_index.append(np.nan)
+
+            (
+                collisions["level_number_lower"],
+                collisions["level_number_upper"],
+                collisions["gi"]
+            ) = (lower_level_index, upper_level_index, gi)
+
+            collisions = collisions.dropna(subset=['level_number_lower', 'level_number_upper'])
+            
+            collisions["atomic_number"] = ion[0]
+            collisions["ion_number"] = ion[1]
+            collisions = collisions.drop(columns=["label_lower", "label_upper"])
+
+            collisions = collisions.astype({
+                'level_number_upper': int,
+                'level_number_lower': int,
+            })
+
+            collisions = collisions.set_index(
+                [
+                    "atomic_number",
+                    "ion_number",
+                    "level_number_lower",
+                    "level_number_upper",
+                ]
+            )
+            # divide the dataframe by gi and remove the column
+            collisions = collisions.iloc[:,:-1].div(collisions.gi, axis=0)
+            collisions.columns = collisions.columns.map(float)
+            
+            col_interpolator.append(interpolate.interp1d(
+                collisions.columns,
+                collisions.values,
+                bounds_error=False,
+                axis=1,
+            ))
+            
+            t_grid.extend(collisions.columns)
+            col_list.append(collisions)
+        
+        if temperature_grid is None:
+            temperature_grid = sorted(list(set(t_grid)))
+
+        col_interp = []
+        for ion_col_data, interpolator in zip(col_list, col_interpolator):
+            ion_interp = pd.DataFrame(
+                interpolator(temperature_grid),
+                index=ion_col_data.index,
+                columns=temperature_grid
+            )
```

Comment:
I think it would be good to have a `logger.info` printout here that states that we are filling in `N` values that are outside the tabulated temperature range using the last valid value.